### PR TITLE
Issue 3165: Bookie does not show logs in 4.15.0rc0

### PR DESCRIPTION
**Pull request by RaulGracia**
### Motivation

Upgrades log4j2 version to fix a problem using the log4j bridge.

### Changes

Upgrade log4j2 version from `2.17.1` to `2.17.2`.

The motivation is that there are multiple issues related to log4j bridge fixed in that minor version: https://logging.apache.org/log4j/2.x/changes-report.html#a2.17.2 

Note that this issue should be cherry-picked to `branch-4.15`.

Master Issue: #3165

### Tests
The problem was introduced in `2.17.1`. This is the result we obtained running a local Bookie with `2.17.1`:

```
bin/bookkeeper localbookie 1
2022-04-06T18:54:18,075 - WARN  - [main:ServerCnxnFactory@309] - maxCnxns is not configured, using default value 0.
2022-04-06T18:54:18,250 - WARN  - [main-SendThread(127.0.0.1:2181):ClientCnxn$SendThread@1279] - An exception was thrown while closing send thread for session 0x10001aa9e8d0000.
org.apache.zookeeper.ClientCnxn$EndOfStreamException: Unable to read additional data from server sessionid 0x10001aa9e8d0000, likely server has closed socket
	at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77) ~[org.apache.zookeeper-zookeeper-3.6.2.jar:3.6.2]
	at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350) ~[org.apache.zookeeper-zookeeper-3.6.2.jar:3.6.2]
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1275) [org.apache.zookeeper-zookeeper-3.6.2.jar:3.6.2]
2022-04-06T18:54:18,905 - ERROR - [main:Journal$LastLogMark@252] - Problems reading from /tmp/bk-data/bookie0/current/lastMark (this is okay if it is the first time starting this bookie
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.bookkeeper.util.NativeIO (file:/home/raul/Documents/workspace/bookkeeper-security-upgrade-new/bookkeeper/bookkeeper-dist/server/target/bookkeeper-server-4.14.3-SNAPSHOT/lib/org.apache.bookkeeper-bookkeeper-server-4.14.3-SNAPSHOT.jar) to field java.io.FileDescriptor.fd
WARNING: Please consider reporting this to the maintainers of org.apache.bookkeeper.util.NativeIO
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

(NO MORE LOGS BEYOND THIS POINT)
```

After applying the current change, we see the expected logs again:
```
bin/bookkeeper localbookie 1
2022-04-06T18:57:05,103 - INFO  - [main:LocalBookKeeper@458] - Using configuration file /home/raul/Documents/workspace/bookkeeper-security-upgrade-new/bookkeeper/bookkeeper-dist/server/target/bookkeeper-server-4.14.3-SNAPSHOT/conf/bk_server.conf
2022-04-06T18:57:05,110 - INFO  - [main:LocalBookKeeper@90] - Running 1 bookie(s) on zk ensemble = '127.0.0.1:2181'.
2022-04-06T18:57:05,112 - INFO  - [main:LocalBookKeeper@120] - Starting ZK server
2022-04-06T18:57:05,117 - INFO  - [main:ZookeeperBanner@42] - 
2022-04-06T18:57:05,117 - INFO  - [main:ZookeeperBanner@42] -   ______                  _                                          
2022-04-06T18:57:05,117 - INFO  - [main:ZookeeperBanner@42] -  |___  /                 | |                                         
2022-04-06T18:57:05,117 - INFO  - [main:ZookeeperBanner@42] -     / /    ___     ___   | | __   ___    ___   _ __     ___   _ __   
2022-04-06T18:57:05,118 - INFO  - [main:ZookeeperBanner@42] -    / /    / _ \   / _ \  | |/ /  / _ \  / _ \ | '_ \   / _ \ | '__|
2022-04-06T18:57:05,118 - INFO  - [main:ZookeeperBanner@42] -   / /__  | (_) | | (_) | |   <  |  __/ |  __/ | |_) | |  __/ | |    
2022-04-06T18:57:05,118 - INFO  - [main:ZookeeperBanner@42] -  /_____|  \___/   \___/  |_|\_\  \___|  \___| | .__/   \___| |_|
2022-04-06T18:57:05,118 - INFO  - [main:ZookeeperBanner@42] -                                               | |                     
2022-04-06T18:57:05,118 - INFO  - [main:ZookeeperBanner@42] -                                               |_|                     
2022-04-06T18:57:05,118 - INFO  - [main:ZookeeperBanner@42] - 
2022-04-06T18:57:05,119 - INFO  - [main:Environment@98] - Server environment:zookeeper.version=3.6.2--803c7f1a12f85978cb049af5e4ef23bd8b688715, built on 09/04/2020 12:44 GMT
2022-04-06T18:57:05,120 - INFO  - [main:Environment@98] - Server environment:host.name=ignatius
2022-04-06T18:57:05,120 - INFO  - [main:Environment@98] - Server environment:java.version=11.0.14.1
2022-04-06T18:57:05,120 - INFO  - [main:Environment@98] - Server environment:java.vendor=Ubuntu
2022-04-06T18:57:05,121 - INFO  - [main:Environment@98] - Server environment:java.home=/usr/lib/jvm/java-11-openjdk-amd64
```

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
Great!

@dlg99 this will unblock the release

**Review by nicoloboschi — APPROVED** [review]

**Comment by RaulGracia** [comment]
@eolivelli please, could you do the same test where you found the problem in RC0 to make sure everything works before merging? Thanks!

**Comment by hangc0276** [comment]
It seams still have some problem. 
When I run bookies in local using the command `bin/bookkeeper localbookie 1`
```
2022-04-07T07:57:38,155 - INFO  - [main:FileTxnSnapLog@479] - Snapshotting: 0x0 to /var/folders/9c/rvr_sgfd6gz7mmnp33j0bfcr0000gn/T/zookeeper13765277709709926628test/version-2/snapshot.0
2022-04-07T07:57:38,156 - INFO  - [main:ZooKeeperServer@547] - Snapshot taken in 2 ms
2022-04-07T07:57:38,171 - INFO  - [ProcessThread(sid:0 cport:2181)::PrepRequestProcessor@138] - PrepRequestProcessor (sid:0) started, reconfigEnabled=false
2022-04-07T07:57:38,171 - INFO  - [main:RequestThrottler@75] - zookeeper.request_throttler.shutdownTimeout = 10000 ms
2022-04-07T07:57:38,199 - INFO  - [NIOWorkerThread-1:FourLetterCommands@223] - The list of known four letter word commands is : [{1936881266=srvr, 1937006964=stat, 2003003491=wchc, 1685417328=dump, 1668445044=crst, 1936880500=srst, 1701738089=envi, 1668247142=conf, -720899=telnet close, 1751217000=hash, 2003003507=wchs, 2003003504=wchp, 1684632179=dirs, 1668247155=cons, 1835955314=mntr, 1769173615=isro, 1920298859=ruok, 1735683435=gtmk, 1937010027=stmk}]
2022-04-07T07:57:38,200 - INFO  - [NIOWorkerThread-1:FourLetterCommands@224] - The list of enabled four letter word commands is : [[wchs, stat, wchp, dirs, stmk, conf, ruok, mntr, srvr, wchc, envi, srst, isro, dump, gtmk, telnet close, crst, hash, cons]]
2022-04-07T07:57:38,200 - INFO  - [NIOWorkerThread-1:NIOServerCnxn@514] - Processing stat command from /127.0.0.1:52434
2022-04-07T07:57:38,205 - INFO  - [NIOWorkerThread-1:StatCommand@52] - Stat command output
2022-04-07T07:57:38,207 - INFO  - [main:LocalBookKeeper@450] - Server UP
```
It won't print any logs after `Server UP` print out.

**Comment by hangc0276** [comment]
When I run unit tests in idea, It doesn't have any logs print out.

**Comment by eolivelli** [comment]
@hangc0276 for tests I started this draft, but these days I don't have time to finish the work, maybe can you help ?
https://github.com/apache/bookkeeper/pull/3148

**Comment by eolivelli** [comment]
@hangc0276 I see this log after "Server UP"
2022-04-07T08:49:01,767 - INFO  - [SyncThread:0:FileTxnLog@285] - Creating new log file: log.1

**Comment by eolivelli** [comment]
I have tested the patch locally. I also started a simple cluster with one booke (bin/bookkeeper zookeeper, bin/bookkeeper bookie....)
Everything works well

I would commit this patch.
We can follow up with the logs in the testing.

**Comment by hangc0276** [comment]
> I have tested the patch locally. I also started a simple cluster with one booke (bin/bookkeeper zookeeper, bin/bookkeeper bookie....) Everything works well
> 
> I would commit this patch. @hangc0276 We can follow up with the logs in the testing.

@eolivelli  OK，Thanks for your reply

**Pull request merged**
